torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -950,6 +968,22 @@ def call_function(
             should_flatten_outputs=True,
         )
 
+        subgraph_example_value = [
+            proxy.node.meta["example_value"] for proxy in body_r.as_proxy()
+        ]
+        with tx.output.fake_mode:
+            map_example_out = [
+                torch.empty_strided(
+                    size=(sample_shape[0], *t.size()),
+                    stride=(0, *t.stride()),
```

Comment:
I'm still perplexed though. Is the output of map really expanded? This is hard for me to believe, because I thought you want to actually map the function on each row individually, so you couldn't possibly have an expanded output.

Reply:
For map, we speculate the computation of one row because the map fn user that writes is written for one row programming-model wise. So we need to trace the subgraph using the sliced input and will re-use the subgraph for other rows. We assume other rows would produce the same output shapes and we just create an output based on single row's result. This would still be correct and can avoid specializing first dim.

Reply:
In plain terms, if I say `ret.stride(0)` in eager, do I get zero or non-zero? Seems to me that I get a non-zero value.

Reply:
Oh yeah! It should be mul(*t.sizes). I don't remember how did I compute the stride hhah.